bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/watson/ElementTreeWriter.java

```diff
@@ -79,11 +79,17 @@ public Object readData(IPath path, DataInput input) {
 	}
 
 	/**
-	 * Sorts the given array of trees and The sort order is written to the given
-	 * output stream.
+	 * Writes indexes of sorted trees to output stream.
+	 *
+	 * @return sorted array of trees.
+	 * @throws IOException if sorting can not be done.
+	 * @see SaveManager#sortTrees
 	 */
-	protected ElementTree[] sortTrees(ElementTree[] trees, DataOutput output) throws IOException {
+	protected ElementTree[] writeSortedTrees(ElementTree[] trees, DataOutput output) throws IOException {
 		ElementTree[] sorted = SaveManager.sortTrees(trees);
+		if (sorted == null) {
```

Comment:
This situation is already logged in sortTrees. Why do you think an exception should be thrown here? Would this result in a popup message in case of an error?
I don't think that throwing an IOException is right here then. The invoked operation is not involving IO. When an exception is desired, raise it by Objects.requireNonNull.

Reply:
OK, thanks for pointing this out.